sync-core/src/main/java/com/cloudant/sync/datastore/PreparedAttachment.java

```diff
@@ -81,16 +93,20 @@ public PreparedAttachment(Attachment attachment,
             IOUtils.closeQuietly(attachmentInStream);
             IOUtils.closeQuietly(tempFileOutStream);
         }
+        
         //Set attachment length from bytes read in input stream
-        this.length = totalRead;
+        if (this.attachment.encoding == Attachment.Encoding.Plain) {
+            this.length = totalRead;
+            // 0 signals "no encoded length" - this is consistent with couch which does not send
+            // encoded_length if the encoding is "plain"
+            this.encodedLength = 0;
```

Comment:
Agreed. The logic is hard to follow but I think we should be consistent with the naming and meaning of these values in CouchDB. So the constant will at least help with that.